motionplan/plannerOptions.go

```diff
@@ -198,18 +206,100 @@ type plannerOptions struct {
 
 	PlanningAlgorithmSettings AlgorithmSettings `json:"planning_algorithm_settings"`
 
-	Fallback *plannerOptions `json:"fallback_options"`
+	RandomSeed int `json:"rseed"`
+
+	PathStepSize float64 `json:"path_step_size"`
+
+	MeshesAsOctrees bool `json:"meshes_as_octrees"`
+
+	Fallback *PlannerOptions `json:"fallback_options"`
 
 	TimeMultipleAfterFindingFirstSolution int `json:"time_multiple_after_finding_first_solution"`
 }
 
+// NewPlannerOptionsFromExtra returns basic default settings updated by overridden parameters
+// found in the "extra" of protobuf MoveRequest. The extra must be converted to an instance of
+// map[string]interface{} first.
+func NewPlannerOptionsFromExtra(extra map[string]interface{}) (*PlannerOptions, error) {
+	opt := NewBasicPlannerOptions()
+
+	jsonString, err := json.Marshal(extra)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(jsonString, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	if opt.CollisionBufferMM < 0 {
+		return nil, errors.New("collision_buffer_mm can't be negative")
+	}
```

Comment:
This feels out of place here.  Is there another place we are doing validation?

Reply:
`validatePlanRequest` maybe? Or I can make a new validate function